internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -177,25 +195,65 @@ func (r *Reconciler) reconcile(ctx context.Context, clusterClass *clusterv1.Clus
 		return ctrl.Result{}, kerrors.NewAggregate(errs)
 	}
 
-	// Ensure the variables are added to the ClusterClass status.
-	clusterClass.Status.Variables = []clusterv1.ClusterClassStatusVariable{}
-	for _, variable := range clusterClass.Spec.Variables {
-		clusterClass.Status.Variables = append(clusterClass.Status.Variables,
-			clusterv1.ClusterClassStatusVariable{
-				Name: variable.Name,
-				Definitions: []clusterv1.ClusterClassStatusVariableDefinition{
-					{
-						From:     clusterv1.VariableDefinitionFromInline,
-						Required: variable.Required,
-						Schema:   variable.Schema,
-					},
-				}})
+	// Reconcile variables
+	if err := r.reconcileVariables(ctx, clusterClass); err != nil {
+		return ctrl.Result{}, err
 	}
+
 	reconcileConditions(clusterClass, outdatedRefs)
 
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) reconcileVariables(ctx context.Context, clusterClass *clusterv1.ClusterClass) error {
```

Comment:
Note: when we implement support for more than one variable definition with the same name we need a unit test for this function alone, eventually injecting something that replace the actual runtime call for test pourposes.

Another option will be to split the call to runtime extension from computing status (before we call all runtime extensions storing results in some temporary struct, then we compute status processing all the results), so we can actually unit test compute status in isolation

Reply:
Added a note linking here to the umbrella issue.